channels/drive/client/drive_file.c

```diff
@@ -90,6 +90,34 @@ static BOOL drive_file_fix_path(WCHAR* path, size_t length)
 	return TRUE;
 }
 
+static BOOL contains_dotdot(const WCHAR* path)
+{
+	WCHAR dotdotbuffer[6] = { 0 };
+	const WCHAR* dotdot = InitializeConstWCharFromUtf8("..", dotdotbuffer, ARRAYSIZE(dotdotbuffer));
+	const WCHAR* tst = path;
+	const size_t len = _wcslen(path);
+
+	do
+	{
+		tst = _wcsstr(tst, dotdot);
+		if (!tst)
+			return FALSE;
+
+		/* Filter .. sequences in file or directory names */
+		if ((*(tst - 1) == L'/') || (*(tst - 1) == L'\\'))
+			return TRUE;
```

Comment:
Shouldn't we check for both - start and end delimiters? Currently i.e. `foo/..foo/bar.txt` would return `TRUE`.

Reply:
hmm, true, need to check that as well.